windowsforms/FlexGridShowcaseDemo/cs/MainForm.cs

```diff
@@ -0,0 +1,704 @@
+            // add footers
+            var footers = _flexGrid.Footers;
+            footers.Descriptions.Add(footerDescription);
+            footers.Fixed = true;
+
+            // set details
+            _flexGrid.RowDetailProvider = (g, r) => new CustomRowDetail();
+
+            // add red style
+            var redStyle = _flexGrid.Styles.Add("Red");
+            redStyle.ImageAlign = ImageAlignEnum.LeftCenter;
+            redStyle.ForeColor = Color.Red;
+
+            // add green style
+            var greenStyle = _flexGrid.Styles.Add("Green");
+            greenStyle.ImageAlign = ImageAlignEnum.LeftCenter;
+            greenStyle.ForeColor = Color.Green;
+
+            // add rating style
+            var ratingStyle = _flexGrid.Styles.Add("Rating");
+            ratingStyle.ImageAlign = ImageAlignEnum.RightCenter;
+        }
+
+        private void InitRules()
+        {
+            _rulesManager.SetC1RulesManager(_flexGrid, _rulesManager);
+            var rulesDict = new Dictionary<string, string>()
+            {
+                { "Discount < 10%", "= [Discount] < 0.1" },
+                { "Discount < 20%", "= [Discount] < 0.2" },
+                { "Discount < 30%", "= [Discount] < 0.3" }
+            };
+
+            // creating rules
+            _rules = rulesDict.Keys
+                .Select(x => new C1.Win.RulesManager.Rule()
+                {
+                    Name = x,
+                    Expression = rulesDict[x],
+                    Style = new ItemStyle()
+                    {
+                        ForeColor = Color.FromArgb(s_rnd.Next(255), s_rnd.Next(255), s_rnd.Next(255)),
+                        BorderColor = Color.DarkBlue,
+                        FontStyle = FontStyle.Bold
+                    }
+                });
+
+            // add menu items
+            var ruleRibbonToggleButtons = rulesDict.Keys
+                .Select(x => new RibbonToggleButton()
+                {
+                    Text = x,
+                    Pressed = false
+                });
+
+            foreach (var ruleRibbonToggleButton in ruleRibbonToggleButtons)
+            {
+                ruleRibbonToggleButton.PressedButtonChanged += RuleRibbonToggleButton_PressedButtonChanged;
+                _ribbonMenuFormatting.Items.Add(ruleRibbonToggleButton);
+            }
+        }
+
+        #endregion
+
+        #region Themes
+
+        private void InitThemes()
+        {
+            // register custom theme
+            var customThemePath = Path.Combine(Directory.GetCurrentDirectory(), CustomThemeName + ".c1themez");
+            if (File.Exists(customThemePath))
+            {
+                C1ThemeController.RegisterTheme(customThemePath);
+            }
+
+            // load themes into ribbon combo box
+            var themes = C1ThemeController.GetThemes().Where(x => x == CustomThemeName || x.Contains("Office2016") || x.Contains("Material"));
+            foreach (var theme in themes)
+            {
```

Comment:
This would be a little more performant:
```suggestion
            foreach (var theme in C1ThemeController.GetThemes())
            {
                if (!(theme == CustomThemeName || theme.Contains("Office2016") || theme.Contains("Material")))
                    continue;
```

Reply:
Fixed with a8a8a722a883c1f1379f5ab2ac9d3b718897848f.